packages/gatsby-remark-images/src/index.js

```diff
@@ -268,12 +268,15 @@ module.exports = (
     `.trim()
     }
 
+    const shouldApplyMaxWidth =
+      !imageCaption && wrapperStyle.indexOf(`max-width:`) !== -1
+
     rawHTML = `
     <span
       class="${imageWrapperClass}"
       style="position: relative; display: block; margin-left: auto; margin-right: auto; ${
         imageCaption ? `` : wrapperStyle
-      } max-width: ${presentationWidth}px;"
+      } ${shouldApplyMaxWidth ? `max-width: ${presentationWidth}px;` : ``}"
```

Comment:
What about moving `max-width` before ``${imageCaption ? `` : wrapperStyle}``? So `maxWidth` from user supplied `wrapperStyle` would overwrite default one.

This seems fragile - we have 2 different conditions relying on `imageCaption` and if we ever change this condition in future, it's very easy to miss it in one of the places